### DISCUSSION TITLE
Logic to create classname could be wrong

### DISCUSSION THREAD

**Discussion by vietduc1441** [cssinjs/jss#1601] [Q&A] [answered]
Hi all,
Now the class name for minimize is calculated by:

${prefix || 'c'}${moduleId}${jssId}${ruleCounter}

In my app, I have many small apps, each has it own jss version, sharing many components. We know that Moduleid will increase 1 each time to one instance of lib is loaded. Now i have some conflicts of class name. So i am thinking the formula could be wrong as follows:

At beginning 
moduleId:1
Jssid:1
RuleCounter:37

then i navigate to other pages many time till one moment:
ModuleId:11
Jssid:3
RuleCounter:7

so both produce c1137 as class name and will be conflicted.

could it be the case?

**Comment by kof** [comment]
Module id is literally the javascript module, it is created once the module is evaluated, I doubt you literally have jss library  loaded several times for each case or do you? 

Jss id is on the other hand something you can create many times, it is incremented with each new jss instance.

**Reply by vietduc1441** [reply]
In my case, i have many independent sub-apps which dont share jss. Each have it own jss library bundled inside its code.

I can see that it could reach to 13 or more by evaluate  this['2f1acc6c3a606b082e5eef5e54414ffb'].
So the situation i described could be happen?

If yes, is there any way as workaround by customize the css. I tried 

`<JssProvider generateId={generateId} classNamePrefix="App1-"> `

But the class name doest change. (I only use styled-jss in my components)

**Reply by kof** [reply]
if you have many evaluated modules, each produces a unique prefix, so that's not the problem

**Comment by vietduc1441** [comment, marked as answer]
Yep, the moduleId is definitely different. But it still could produce 2 identical class name:
c.1.1.137
c.11.1.37

**Reply by kof** [reply]
but module id is part of the class name, how can class name be identical if module id is different

**Reply by vietduc1441** [reply]
In this example 
c.1.1.137
c.11.1.37
So the moduleId is 1 and 11 but class name could be the same because the next parts make the total the same.

**Reply by kof** [reply]
oh, because we don't have a separator character in the class name, interesting find

**Reply by kof** [reply]
We need to think of some way making sure this doesn't happen, its clearly an oversight. We wanted to have very short class names for production but didn't realize how this can go wrong.

I think we need to put the "-" back into the production class name too. Wanna create a PR?

**Reply by vietduc1441** [reply]
Yes, i will create one.
